2.6/root/usr/share/container-scripts/mongodb/replset-functions.sh

```diff
@@ -0,0 +1,91 @@
+# cache_container_addr waits till the container gets the external IP address and
+# cache it to disk
+function cache_container_addr() {
+  echo -n "=> Waiting for container IP address ..."
+  for i in $(seq $MAX_ATTEMPTS); do
+    result=$(ip addr | grep 'state UP' -A2 | tail -n1 | awk '{print $2}' | cut -f1 -d'/')
+    if [ ! -z "${result}" ]; then
+      echo -n $result > ${HOME}/.address
+      echo " $(mongo_addr)"
+      return 0
+    fi
+    sleep $SLEEP_TIME
+  done
+  echo "Failed to get Docker container IP address." && exit 1
+}
+
+# container_addr returns the current container external IP address
+function container_addr() {
+  echo -n $(cat ${HOME}/.address)
+}
+
+# mongo_addr returns the IP:PORT of the currently running MongoDB instance
+function mongo_addr() {
+  echo -n "$(container_addr):${port}"
+}
+
+# endpoints returns list of IP addresses with other instances of MongoDB
+# To get list of endpoints, you need to have headless Service named 'mongodb'.
+# NOTE: This won't work with standalone Docker container.
+function endpoints() {
+  service_name=${MONGODB_SERVICE_NAME:-mongodb}
+  dig ${service_name} A +search +short 2>/dev/null
+}
+
+# replset_addr return the address of the current replSet
+# NOTE: this function hopes that some replset member use standart port
+function replset_addr() {
+  echo "${MONGODB_REPLICA_NAME}/$(echo $(endpoints) | sed -e 's| |:27017,|g' -e 's|$|:27017|')"
+}
+
+# mongo_wait_primary waits until primary replset member is ready
+# (accepts connection and master is elected)
+# $1 - login parameters for mongo (optional)
+# $2 - host where to connect (replset_addr by default)
+function mongo_wait_replset() {
+  # If there is no PRIMARY yet, rs.isMaster().primary returns "undefined"
+  for i in $(seq 1$MAX_ATTEMPTS); do
+    # Test connection to replica set
+    local set=$(mongo admin ${1:-} --host ${2:-$(replset_addr)} --quiet --eval "rs.isMaster().setName;" | tail -n 1)
+    # If there is no PRIMARY yet, rs.isMaster().primary returns "undefined"
+    local primary=$(mongo admin ${1:-} --host ${2:-$(replset_addr)} --quiet --eval "rs.isMaster().primary;" | tail -n 1)
```

Comment:
`local` declarations and assignments should go in separate lines, specially when there's a subshell / evaluation on the right side. Here is the explanation why: https://google.github.io/styleguide/shell.xml?showone=Use_Local_Variables#Use_Local_Variables